lib/core/test/unit/Cardano/Wallet/DB/Properties.hs

```diff
@@ -250,7 +266,13 @@ sortedUnions = map (Identity . sort' . runIdentity) . unions
 
 -- | Execute an action once per key @k@ present in the given list
 once :: (Ord k, Monad m) => [(k,v)] -> ((k,v) -> m a) -> m [a]
-once xs = forM (Map.toList (Map.fromList xs))
+once xs action = fmap catMaybes $ flip evalStateT mempty $
+    forM xs $ \(k, v) -> do
+        s <- get
+        modify' (Set.insert k)
+        case Set.member k s of
```

Comment:
almost sure hlint will shout here and force you to use `if ... then ... else` :-)

Reply:
It's weird... I had quite a few problems with hlint today, seems like my editor doesn't run it anymore.. need to look into that.